src/cli.rs

```diff
@@ -148,13 +148,14 @@ impl Rover {
                 match self.output_type {
                     OutputType::Plain => output.print()?,
                     OutputType::Json => JsonOutput::from(output).print()?,
+                    OutputType::Markdown => output.print_markdown()?,
                 }
                 process::exit(0);
             }
             Err(error) => {
                 match self.output_type {
                     OutputType::Json => JsonOutput::from(error).print()?,
-                    OutputType::Plain => {
```

Comment:
i think what we'll want here is a `fn print_markdown(&self) -> Result<()>` in the `impl RoverError` block that does pretty much exactly what you already did in the `impl RoverOutput` block. i'm thinking the output might look something like this:

```
### Check Failed

#### Reason

The main error text

#### Suggestion

The suggestion text
```

the code you'll have to update for this can be found [here](https://github.com/apollographql/rover/blob/main/src/error/mod.rs#L81)

and you'll likely have to refactor some stuff out of the [`impl Display for RoverError` block](https://github.com/apollographql/rover/blob/main/src/error/mod.rs#L114) in order to construct the markdown properly. you'll also want to make sure to handle the [`RoverClientError::OperationCheckFailure`](https://github.com/apollographql/rover/blob/main/src/error/mod.rs#L95) specifically and print out a markdown formatted table for it. (this may also be worth a refactor to the `get_table` function to take an `is_markdown: bool`.)

---

Reply:
> (this may also be worth a refactor to the get_table function to take an is_markdown: bool.)

Since default parameter doesn't exist in Rust, I'm not in the bandwagon to "destroy" current API. the best that I can do is to pass a `struct` with a `Default` impl but it feels heavy just for this usecase.

Reply:
> Since default parameter doesn't exist in Rust, I'm not in the bandwagon to "destroy" current API.

that "API" is only ever used internally, so it _should_ be fine to "break" it. The biggest thing we worry about with regards to breakage here is _external_ usage (like, no longer accepting a specific argument that we used to support)